.github/workflows/build.yml

```diff
@@ -275,6 +275,11 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
+      - name: Setup Swift
+        uses: swift-actions/setup-swift@v1.18.0
+        with:
+          swift-version: "5.7"
```

Comment:
I would advise using the build containers over building an environment with GHA for every CI job (as it was, in the past).  Builds will take longer if you have to pull in every language like this, as opposed to downloading a one-time generated build container with everything in it.  With a build container anyone can use it to build the entire package locally and the local build is the same as the CI build.  With GHA scripted build environments like this, only GHA can run the build.

Reply:
I agree I like the old Travis way of having a docker container with everything setup in it. I'm not sure if you can do that on GitHub actions but I'd support that

Reply:
GitHub supports [hosted runners][1] but that requires infra work from apache i guess

[1]: https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners

Reply:
if we only run on GitHub hosted runners, then the only way is to follow the current approach: [adding steps to customize the environment][1]

[1]: https://docs.github.com/en/actions/using-github-hosted-runners/customizing-github-hosted-runners